### DISCUSSION TITLE
Buildings Library: Nothing but errors

### DISCUSSION THREAD

**Discussion by NicholasFry** [OpenModelica/OpenModelica#9995] [Q&A]
Hi,

I am trying to run some of the examples in the LBNL buildings library. This is my second attempt at trying to learn to use OM (not code Modelica). If I simply try to run the packaged examples, I get a "Translation Error." There are no "Translation Error" explanations in the docs and no relevant topics in discussion. For example:
"component rooVol contains the definition of a partial class Medium.
Please redeclare it to any package compatible with Modelica.Media.Interfaces.PartialMedium."

What is this? What does it mean? How do I fix it?

Background:
User only and always. 

Version:
OpenModelica v1.20.0 (64-bit)
OMSimulator v2.1.1.post188-gaf996ad-mingw

Thank you!

**Comment by casella** [comment]
@NicholasFry what do you mean exactly as "packaged examples"? Can you name some of the models that are giving you trouble? According to our [test reports](https://libraries.openmodelica.org/branches/v1.20/Buildings_latest/Buildings_latest.html), most of the _runnable_ models in the Buildings library should work out of the box with 1.20.0. 

Maybe you are trying to run _components_ which are not fully parameterized. The error you get refers to components handling fluids that need their Medium model to be redeclared. When you build new system models, you can do that by enabling replaceable support in Tools | Options | General; then, a drop-down menu will appear in the parameter window where you can select the Medium model you want to use.

Hope this helps 😃

**Reply by NicholasFry** [reply]
I assume this is what you are talking about?
![image](https://user-images.githubusercontent.com/60554125/210785062-634f2db1-a8d6-485c-ba38-fdd68e593119.png)

I am now running into a definition problem in the scope for a component - mass flow. I did copy and paste this component from an example. How do I reinstantiate this into the scope for the .mo script? I assume that is what needs to happen.
![image](https://user-images.githubusercontent.com/60554125/210785769-e389feb4-d6bf-45cf-b26e-7502ebc3967d.png)

I am also running into the problem of assigning Medium 1 and 2 across the whole system. I referred to a 2020 post on SO - https://stackoverflow.com/questions/64385558/how-to-specify-medium-in-openmodelica
"Unfortunately, currently there is no possibility of choosing working fluid in OpenModelica via Diagram View. You should refer to Text View and add replaceable package Medium =       Modelica.Media.Water.StandardWater      constrainedby Modelica.Media.Interfaces.PartialMedium; or any other Medium of your interest. Then, under each of your component that contains this Medium you should add the following line: redeclare package Medium = Medium,. Available working fluids you can find under Modelica.Media"

Is this still the case or can this now be done in the diagram view? What appears now is a cluster of junk errors.
![image](https://user-images.githubusercontent.com/60554125/210789527-b2141c4f-a348-4a31-a88d-0463da677d4c.png)

**Comment by NicholasFry** [comment]
There is not much more that I can say about the examples, other than they look like this:
![image](https://user-images.githubusercontent.com/60554125/210455712-7e588ba0-6805-4617-adc8-1a266143a9e3.png)

Today they seem to work for the most part. I have a feeling that they failed to function because I was initially trying to build my own models. The medium errors did not appear today using the packaged examples.

**Comment by casella** [comment]
If you build your own, please make sure you always select the medium model for all fluid components.

**Reply by NicholasFry** [reply]
I know this is not your library, but is there a simple example of buildings connected to boreholes? It seems all the examples I am finding have very novel applications, like ceiling radiant, but no heating AND cooling load connected to a borehole array. I am trying to get my mechanical engineering counterparts to invest some time in this but if I cannot build and run a simple model, I will not convince them of Modelica's worth.

**Reply by mwetter** [reply]
@NicholasFry : The model https://simulationresearch.lbl.gov/modelica/releases/v9.1.0/help/Buildings_Experimental_DHC_Examples_Combined.html#Buildings.Experimental.DHC.Examples.Combined.SeriesVariableFlow has heating and cooling load connected to a borehole array.

PS: I opened an issue to avoid being able to select glycol for the `MassFlowSource_WeatherData` model to avoid such a confusing choice in the future (https://github.com/ibpsa/modelica-ibpsa/issues/1681)

**Reply by NicholasFry** [reply]
That looks a lot different than I would expect. These are the reduced order models you reference often. Can I ask if you have more documentation on the ETS? I am not sure I fully understand the difference yet between the buildings and partial buildings or the ETS and the partial ETS, for example. That might be quite simple. Is this reference to a part or full load?

Are there any examples of EP Spawn profiles being entered for the Partial ETS building definition? This would make sense at the utility scale that we are looking for. This would also be a deviation from workflows in my current practice. We would have to be able to enter 8760s from a cluster of buildings or a large building for the partial ETS from CAD-based EP outputs.

The wastewater energy transfer component looks great though.

**Reply by mwetter** [reply]
@NicholasFry : The ETS that are used in SeriesVariableFlow are described in https://doi.org/10.1016/j.energy.2020.117418 which uses an earlier model of that system.

See for example `Buildings.Experimental.DHC.Loads.BaseClasses.Examples` for how Spawn models will be coupled to ETS. @dhblum may be able to point you to larger examples models that are on development branches, and https://doi.org/10.1016/j.apenergy.2022.119880 also uses a mix of time series models and a Spawn model coupled to a geothermal DHC system.

**Comment by casella** [comment]
I'd suggest you to contact @mwetter (mwetter@lbl.gov), who is the leader of the development, and ask for advice.
